### PR TITLE
Home Page redesign

### LINKED ISSUE #37: Update fonts and colors
Once the styling updates in #33 have been finalized in Figma, we can start incorporating those changes by updating the font sizes and colors.

### LINKED ISSUE #65: Development work for Home page
This is a placeholder for the development work that will take place this quarter. Issues will be built out once designs and content are done.

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #65 

## What this PR does

- Implements new design for Home Page, Footer
- Adds design fixes to: Job Posting page, some basic alignment to Jobs page

###  Home Page
- Everything except for the nav (#44) is implemented
image
image
image

### Layouts
- Adds the ability to declare `classes` on the `<body>` element from a `layout`
- Adds the `posting` class to the `<body>` element on job post pages

### Job Postings
- Correct the margin-top and margin-bottom of the `main` element on the page, for both Desktop and Mobile.
- Add the correct margin/padding bottom to the `<h1>` on the page. 
- Declare the `<h2>`s on the page to actually use _styles_ from `.h3`, so it matches the designs. 

### Jobs
- Add the same row/container/col-width as the Jobs page
- Don't change anything else (Happening in #82)

### CSS refactors
- Adds CSS Variables for colors
- Adds overrides Bootstrap's CSS Variables https://getbootstrap.com/docs/5.2/components/buttons/#variables for `btn`, `btn-primary`, `btn-outline-primary`
- Start the Style Guide

- #### Headlines
- Create _classes_ for `.h1`-`.h5`, along with regular style declarations for `h1`-`h5` elements.
- Use `calc(##rem / 16)` style for font sizes

#### Font families
- Declare overrides for certain Bootstrap CSS Variables: `bs-font-sans-serif`, `bs-font-monospace`
- Correct spelling of a class - It's `sans-serif` not `san-serif`. _Sans_ means "without" in French. So it's "without serifs".

#### Colors
- Adds Bootstrap overrides for `body` colors, `border-light`
- Removed colors not used
- Teal is now called Cyan

Things that actually changed colors from this PR, aside from variables:
- Background color is now darker. Color changed from `#262626` to `#1c1c1c`
- Text on the homepage changed from `#01B4CB` to `#51B1C7`

### Buttons
- `btn btn-primary`
- `btn btn-primary disabled`
- `btn btn-outline-primary`
- `btn btn-outline-primary disabled`
image

- `btn btn-primary` in focus

image

### CODE REVIEW

File: styles/base.css [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-    --navy: #223061;
-    --off-black: #262626;
-    --black: #000000;
-    --grey: #EDEDED;
-    --white: #FFFFFF;
+    --brand-primary-green: #88B440;
+    --brand-primary-cyan: #51B1C7;
+    --brand-primary-black: #1C1C1C;
+    --brand-primary-gray: #ECEDED;
+    --brand-primary-white: #FFFFFF;
```
Comment (machikoyasuda) [inline styles/base.css:8]: From https://www.figma.com/file/2gq3WJFdRFDefyHcgFMUGt/Current-compiler.la?type=design&node-id=1358-19464&mode=dev

File: styles/base.css [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+    --brand-scales-gray-4: #545454;
+    --text-black: #1C1C1C;
```
Comment (machikoyasuda) [inline styles/base.css:10]: Colors from https://www.figma.com/file/2gq3WJFdRFDefyHcgFMUGt/Current-compiler.la?type=design&node-id=1358-19464&mode=dev

File: styles/base.css [hunk 3]
```diff
@@ -34,13 +37,8 @@ body {
 body {
     display: flex;
     flex-direction: column;
-
+    font-family: 'Roboto', Arial, sans-serif;
```
Comment (machikoyasuda) [inline styles/base.css:40]: This will be removed with #94

**Comment by machikoyasuda** [comment]
I decided it was easier to review this all in 1 PR, rather than scattered across 4-5 PRs :)

**Review by thekaveman — CHANGES_REQUESTED** [review]
> I decided it was easier to review this all in 1 PR, rather than scattered across 4-5 PRs :)

I disagree. This PR is massive and focuses on way more than the Home page / colors. 

#82 is our priority for right now. I'm not sure why #49 was folded in here as well, that one was not prioritized for dev work on the Project board. 

Please focus PRs on one issue at a time.

**Pull request closed without merging**